crates/hyperqueue/src/client/output/cli.rs

```diff
@@ -534,6 +535,66 @@ impl Output for CliOutput {
         );
     }
 
+    fn print_job_cat(&self, job: JobDetail, stdout: bool, stderr: bool) {
+        let JobDetail {
+            info,
+            job_desc,
+            tasks,
+            max_fails: _,
+            submission_date,
+            completion_date_or_now,
+            submit_dir,
+        } = job;
+        let JobDescription::Array { ids, .. } = &job_desc;
+        let (resources, program_def, time_limit, priority, pin) = match &job_desc {
+            JobDescription::Array { task_desc, .. } => (
+                &task_desc.resources,
+                &task_desc.program,
+                task_desc.time_limit,
+                task_desc.priority,
+                task_desc.pin,
+            ),
+        };
+
+        let stdout_path = match &program_def.stdout {
+            StdioDef::File(file) => Some(file),
+            _ => None,
+        };
+        if stdout_path.is_none() && stdout {
+            log::error!("Job doesn't have stdout task files, consider submitting job with --stdout");
+            return;
+        }
+
+        let stderr_path = match &program_def.stderr {
+            StdioDef::File(file) => Some(file),
+            _ => None,
+        };
+        if stderr_path.is_none() && stderr {
+            log::error!("Job doesn't have stderr task files, consider submitting job with --stderr");
+            return;
+        }
+
+        for id in ids.iter() {
+            println!("# Task {}", id);
+
+            if stdout {
+                println!("# Stdout");
+                let str_path = stdout_path.unwrap().to_str().unwrap().replace("%{TASK_ID}", &id.to_string());
+                let stream = fs::read_to_string(str_path)
```

Comment:
The data might be binary, so we cannot operate with `str`/`String`. Also we cannot afford to load the data into memory, it might be larger than RAM. We need to stream the file to `stdout` effectively. You can use [`io::copy`](https://static.rust-lang.org/doc/master/std/io/fn.copy.html) for this.